sequencer/src/header.rs

```diff
@@ -245,10 +245,22 @@ impl BlockHeader<SeqTypes> for Header {
         // Fetch the latest L1 snapshot.
         let l1_snapshot = instance_state.l1_client().snapshot().await;
         // Fetch the new L1 deposits between parent and current finalized L1 block.
-        let l1_deposits = fetch_fee_receipts(
-            parent_leaf.get_block_header().l1_finalized,
-            l1_snapshot.finalized,
-        );
+        let l1_deposits = instance_state
+            .l1_client
+            .get_finalized_deposits(
+                parent_leaf
+                    .get_block_header()
+                    .l1_finalized
+                    .map(|block_info| block_info.number),
+                l1_snapshot
+                    .finalized
+                    .map(|block_info| block_info.number)
+                    .unwrap_or(0), // I'm not sure its possible for
```

Comment:
I'm not certain that using 0 as the default block height makes sense, shouldn't we panic? Or at least use the parent's l1 snapshot height?

Reply:
Fixed in  5b475417967703621527409fd234ee3a4f26dda8